ledger-service/http-json/src/main/scala/com/digitalasset/http/PackageService.scala

```diff
@@ -167,37 +174,47 @@ private class PackageService(
   def resolveContractTypeId(implicit ec: ExecutionContext): ResolveContractTypeId =
     resolveContractTypeIdFromState { () =>
       val st = state
-      (st.templateIdMap, st.interfaceIdMap)
+      (st.templateIdMap, st.interfaceIdMap, st.packageLanguageVersionMap)
     }
 
   private[this] def resolveContractTypeIdFromState(
-      latestMaps: () => (TemplateIdMap, InterfaceIdMap)
+      latestMaps: () => (TemplateIdMap, InterfaceIdMap, PackageLanguageVersionMap)
   )(implicit ec: ExecutionContext): ResolveContractTypeId = new ResolveContractTypeId {
     import ResolveContractTypeId.{Overload => O}, domain.{ContractTypeId => C}
     override def apply[U, R](jwt: Jwt, ledgerId: LedgerApiDomain.LedgerId)(
         x: U with ContractTypeId.OptionalPkg
     )(implicit
         lc: LoggingContextOf[InstanceUUID],
         overload: O[U, R],
-    ): Future[Error \/ Option[R]] = {
-      type ResultType = Option[R]
+    ): Future[Error \/ Option[(R, LanguageVersion)]] = {
+      type ResultType = Option[(R, LanguageVersion)]
+      def withLanguageVersionT(
+          r: ResolvedOf[ContractTypeId.Template]
+      ): Option[(ResolvedOf[ContractTypeId.Template], LanguageVersion)] =
+        latestMaps()._3.get(r.packageId).map(l => (r, l))
+      def withLanguageVersionI(
+          r: ResolvedOf[ContractTypeId.Interface]
+      ): Option[(ResolvedOf[ContractTypeId.Interface], LanguageVersion)] =
+        latestMaps()._3.get(r.packageId).map(l => (r, l))
```

Comment:
you can use on method with parametric type for both no?
```suggestion
      def withLanguageVersion[T[_]](r: ResolvedOf[T]): Option[(ResolvedOf[T], LanguageVersion)] =
        latestMaps()._3.get(r.packageId).map(l => (r, l))
```